com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -506,40 +506,41 @@ static void RenderCameraStack(ScriptableRenderContext context, Camera baseCamera
             anyPostProcessingEnabled &= SystemInfo.graphicsDeviceType != GraphicsDeviceType.OpenGLES2;
 
             bool isStackedRendering = lastActiveOverlayCameraIndex != -1;
-            using (new ProfilingScope(null, Profiling.Pipeline.beginCameraRendering))
-            {
-                BeginCameraRendering(context, baseCamera);
-            }
-            // Update volumeframework before initializing additional camera data
-            UpdateVolumeFramework(baseCamera, baseCameraAdditionalData);
-            InitializeCameraData(baseCamera, baseCameraAdditionalData, !isStackedRendering, out var baseCameraData);
 
 #if ENABLE_VR && ENABLE_XR_MODULE
-            var originalTargetDesc = baseCameraData.cameraTargetDescriptor;
             var xrActive = false;
-            var xrPasses = m_XRSystem.SetupFrame(baseCameraData);
+            var xrRendering = true;
+            if (baseCameraAdditionalData != null)
+                xrRendering = baseCameraAdditionalData.allowXRRendering;
+            var xrPasses = m_XRSystem.SetupFrame(baseCamera, xrRendering);
             foreach (XRPass xrPass in xrPasses)
             {
-                baseCameraData.xr = xrPass;
 
-                // XRTODO: remove isStereoEnabled in 2021.x
+                if (xrPass.enabled)
+                {
+                    xrActive = true;
+                    UpdateCameraStereoMatrices(baseCamera, xrPass);
+                }
+
+                using (new ProfilingScope(null, Profiling.Pipeline.beginCameraRendering))
+                {
+                    BeginCameraRendering(context, baseCamera);
+                }
+                // Update volumeframework before initializing additional camera data
+                UpdateVolumeFramework(baseCamera, baseCameraAdditionalData);
+                InitializeCameraData(baseCamera, baseCameraAdditionalData, !isStackedRendering, out var baseCameraData);
+                RenderTextureDescriptor originalTargetDesc = baseCameraData.cameraTargetDescriptor;
+
+                if (xrPass.enabled)
+                {
+                    baseCameraData.xr = xrPass;
+                    // XRTODO: remove isStereoEnabled in 2021.x
```

Comment:
You have to make it obsolete for at least one version. Then in next one remove it.

Reply:
we already marked it obsoltete since URP10 :)